src/Data/Argonaut/Decode/Generic/Rep.purs

```diff
@@ -87,3 +91,32 @@ genericDecodeJson = map Rep.to <<< decodeRep
 
 mFail :: forall a. String -> Maybe a -> Either String a
 mFail msg = maybe (Left msg) Right
+
+-- | A function for decoding `Generic` sum types using string literal representations
+-- | Takes a function for transforming the tag name in encoding
+decodeLiteralSum :: forall a r. Rep.Generic a r => DecodeLiteral r => (String -> String) -> Json -> Either String a
+decodeLiteralSum tagNameTransform = map Rep.to <<< decodeLiteral tagNameTransform
```

Comment:
Do you think it makes sense to provide a transform? I'd keep it simple and drop that. If you're using generics you already can't really control the format, so providing something to customise it for this particular case seems a little odd.